sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1873,6 +1873,8 @@ object AnsiCast {
 
     case (NullType, _) => true
 
+    case (_, StringType) => true
```

Comment:
Does this affect the coming year-month and day-time interval?

Reply:
Yes

Reply:
@cloud-fan . Usually, we do the explicitly allowed-list approach in case of types. Is this change okay?
If this PR aims for `complex type` only, why don't we add them explicitly instead of doing this widely.

Reply:
Also, cc @MaxGekk

Reply:
`df.show` needs to cast the column to string, I think we need to support casting from all the data types here, otherwise `df.show` may still be broken under some cases.

Reply:
So far, we don't support such casting. I opened the JIRAs for that: SPARK-34667 and SPARK-34668